File: frontend/src/features/career-path/types/careerPath.ts
export type CareerPathPriority = "high" | "medium" | "low";

export type CareerPathDifficulty = "easy" | "medium" | "hard";

export type CareerPathGenerateRequest = {
  language?: string;
  target_role: string;
  current_level?: string | null;
  timeframe_months?: number | null;
};

export type CareerPathMilestone = {
  title: string;
  description: string;
  timeframe: string;
  priority: CareerPathPriority | string;
  tasks: string[];
};

export type CareerPathSkillGap = {
  skill: string;
  current_level: string;
  target_level: string;
  reason: string;
  priority: CareerPathPriority | string;
};

export type CareerPathLearningItem = {
  title: string;
  description: string;
  type: string;
  estimated_time: string;
  priority: CareerPathPriority | string;
};

export type CareerPathProjectItem = {
  title: string;
  description: string;
  skills_practiced: string[];
  portfolio_value: string;
  difficulty: CareerPathDifficulty | string;
};

export type CareerPathApplicationStrategyItem = {
  title: string;
  description: string;
  action_items: string[];
};

export type CareerPathInputSnapshot = {
  target_role?: string;
  current_level?: string | null;
  timeframe_months?: number | null;
};

export type CareerPath = {
  id: number;
  user_id: number;

  target_role: string;
  current_level?: string | null;
  timeframe_months?: number | null;

  input_snapshot?: CareerPathInputSnapshot | null;

  roadmap: CareerPathMilestone[];
  skill_gaps: CareerPathSkillGap[];
  learning_plan: CareerPathLearningItem[];
  project_plan: CareerPathProjectItem[];
  application_strategy: CareerPathApplicationStrategyItem[];

  summary?: string | null;
  confidence_score?: number | null;
  role_fit?: "high" | "medium" | "low" | "very_low" | string | null;
  role_fit_summary?: string | null;
  status: string;

  created_at: string;
  updated_at: string;
};

File: backend/app/schemas/career_path/career_path.py
from datetime import datetime
from typing import Any, Optional

from pydantic import BaseModel, Field


class CareerPathGenerateRequest(BaseModel):
    language: str = Field(default="english", max_length=20)
    target_role: str = Field(..., min_length=2, max_length=120)
    current_level: Optional[str] = Field(default=None, max_length=80)
    timeframe_months: Optional[int] = Field(default=6, ge=1, le=36)


class CareerPathMilestone(BaseModel):
    title: str
    description: str
    timeframe: str
    priority: str
    tasks: list[str] = Field(default_factory=list)


class CareerPathSkillGap(BaseModel):
    skill: str
    current_level: str
    target_level: str
    reason: str
    priority: str


class CareerPathLearningItem(BaseModel):
    title: str
    description: str
    type: str
    estimated_time: str
    priority: str


class CareerPathProjectItem(BaseModel):
    title: str
    description: str
    skills_practiced: list[str] = Field(default_factory=list)
    portfolio_value: str
    difficulty: str


class CareerPathApplicationStrategyItem(BaseModel):
    title: str
    description: str
    action_items: list[str] = Field(default_factory=list)


class CareerPathResponse(BaseModel):
    id: int
    user_id: int

    target_role: str
    current_level: Optional[str] = None
    timeframe_months: Optional[int] = None

    input_snapshot: Optional[dict[str, Any]] = None

    roadmap: list[dict[str, Any]] = Field(default_factory=list)
    skill_gaps: list[dict[str, Any]] = Field(default_factory=list)
    learning_plan: list[dict[str, Any]] = Field(default_factory=list)
    project_plan: list[dict[str, Any]] = Field(default_factory=list)
    application_strategy: list[dict[str, Any]] = Field(default_factory=list)

    summary: Optional[str] = None
    confidence_score: Optional[int] = None

    role_fit: Optional[str] = None
    role_fit_summary: Optional[str] = None

    status: str

    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
